src/session-pool.ts

```diff
@@ -247,6 +247,17 @@ export function isInstanceNotFoundError(
   );
 }
 
+/**
+ * Checks whether the given error is an 'Create session permission' error.
+ * @param {Error} error The error to check.
+ * @return {boolean} True if the error is an 'Create session permission' error, and otherwise false.
+ */
+export function isCreateSessionPermissionError(
+  error: grpc.ServiceError | undefined
+): boolean {
+  return error !== undefined && error.code === grpc.status.PERMISSION_DENIED;
```

Comment:
This is only checking for the error code `PERMISSION_DENIED`, and not specifically for permission denied for session creation. The function should also specifically check that it is session creation that was denied (or the name of the function should be changed, but I don't think that is what we want here).

Reply:
This is in catch of `_fill` method and it has only `_createSessions` method. so only Session creation permission will only be here.

Reply:
Even if it is only used in that specific function at this moment, the function is exported and its name does not correspond with what it is actually doing. I would expect `export function isCreateSessionPermissionError(...)` to return false if I fed it a `PERMISSION_DENIED` error for creating for example a database, but it currently returns true.

Reply:
Added Extra condition for message that should include `spanner.sessions.create`.